sources/web/datalab/polymer/components/datalab-sessions/datalab-sessions.ts

```diff
@@ -103,25 +94,31 @@ class SessionsElement extends Polymer.Element {
    * updates the _sessionList property.
    */
   _fetchSessionList() {
-    const self = this;
-    self._fetching = true;
+    // Don't overlap fetch requests. This can happen because we set up fetch from several sources:
+    // - Initialization in the ready() event handler.
+    // - Refresh mechanism called by the setInterval().
+    // - User clicking refresh button.
+    // - Sessions page gaining focus.
+    if (this._fetching) {
+      return Promise.resolve(null);
```

Comment:
I see. Yea I guess we don't need it to then-able currently.
I'll take it out for now, although we might need that for tests later.